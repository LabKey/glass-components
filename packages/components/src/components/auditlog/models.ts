/*
 * Copyright (c) 2016-2018 LabKey Corporation. All rights reserved. No portion of this work may be reproduced in
 * any form or by any electronic or mechanical means without written permission from LabKey Corporation.
 */
import { fromJS, Map, Record } from 'immutable';

export class AuditDetailsModel extends Record({
    rowId: undefined,
    comment: undefined,
    eventUserId: undefined,
    eventDateFormatted: undefined,
    oldData: undefined,
    newData: undefined,
}) {
    rowId?: number;
    comment?: string;
    eventUserId?: number;
    eventDateFormatted?: string;
    oldData?: Map<string, string>;
    newData?: Map<string, string>;

    static create(raw: any): AuditDetailsModel {
        return new AuditDetailsModel({
            ...raw,
            oldData: raw.oldData ? fromJS(raw.oldData) : undefined,
            newData: raw.newData ? fromJS(raw.newData) : undefined,
        });
    }

    getActionLabel = (): string => {
        if (this.isUpdate()) return 'Updated';
        else if (this.isInsert()) return 'Created';
        else if (this.isDelete()) return 'Deleted';
        else return 'Updated';
    };

    isDelete = (): boolean => this.oldData && this.newData && this.oldData.size > 0 && this.newData.size === 0;

    isInsert = (): boolean => this.oldData && this.newData && this.oldData.size === 0 && this.newData.size > 0;

    isUpdate = (): boolean => this.oldData && this.newData && this.oldData.size > 0 && this.newData.size > 0;
}
