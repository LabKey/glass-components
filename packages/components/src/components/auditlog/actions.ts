/*
 * Copyright (c) 2016-2018 LabKey Corporation. All rights reserved. No portion of this work may be reproduced in
 * any form or by any electronic or mechanical means without written permission from LabKey Corporation.
 */
import { Ajax, Utils, ActionURL } from '@labkey/api';

import { AuditDetailsModel } from './models';

export function getAuditDetail(auditRowId: number, auditEventType: string): Promise<AuditDetailsModel> {
    return new Promise((resolve, reject) => {
        Ajax.request({
            url: ActionURL.buildURL('audit', 'GetDetailedAuditChanges.api'),
            method: 'GET',
            params: { auditRowId, auditEventType },
            success: Utils.getCallbackWrapper(response => {
                resolve(AuditDetailsModel.create(response));
            }),
            failure: Utils.getCallbackWrapper(error => {
                console.error('Problem retrieving the audit details', error);
                reject('There was a problem retrieving the audit details.');
            }),
        });
    });
}
