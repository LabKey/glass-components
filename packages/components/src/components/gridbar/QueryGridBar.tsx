/*
 * Copyright (c) 2019 LabKey Corporation
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import React from 'reactn';
import { Map } from 'immutable';

import { ChartSelector } from '../chart/ChartSelector';

import { QueryGridModel } from '../base/models/model';

import { Export } from './Export';
import { QueryGridPaging } from './QueryGridPaging';
import { ViewSelector } from './ViewSelector';
import { URLBox } from './URLBox';
import { GridSelectionBanner } from './GridSelectionBanner';
import { PageSizeSelector } from './PageSizeSelector';

type QueryGridBarButtonResolver = (model?: QueryGridModel) => React.ReactNode;
export type QueryGridBarButtons = React.ReactNode | QueryGridBarButtonResolver;

interface QueryGridBarProps {
    buttons?: QueryGridBarButtons;
    model: QueryGridModel;
    showSampleComparisonReports?: boolean;
    onReportClicked?: Function;
    onCreateReportClicked?: Function;
    onSelectionChange?: (model: QueryGridModel, row: Map<string, any>, checked: boolean) => any;
    advancedExportOption?: Record<string, any>;
}

/**
 * Displays a bar of controls for a query grid based on the properties of the provided model.
 * This includes
 * - a URLBox for use in filtering and sorting (based on model.showSearchBox),
 * - a paging element (based on model.isPaged)
 * - an export menu (always)
 * - a chart selector (based on model.showChartSelector)
 * - a view selector (based on model.showViewSelector)
 * You may also provide a set of buttons to be displayed within the bar.
 */
export class QueryGridBar extends React.PureComponent<QueryGridBarProps, any> {
    render() {
        const {
            buttons,
            model,
            showSampleComparisonReports,
            onReportClicked,
            onCreateReportClicked,
            onSelectionChange,
            advancedExportOption,
        } = this.props;
        let buttonsNode = typeof buttons === 'function' ? (buttons as QueryGridBarButtonResolver)(model) : buttons;

        if (buttons) {
            buttonsNode = <div className="btn-group gridbar-buttons">{buttonsNode}</div>;
        }

        const box = model && model.showSearchBox ? <URLBox key={model.getId()} queryModel={model} /> : null;

        const paging = model && model.isPaged ? <QueryGridPaging model={model} /> : null;

        const pageSizeBtn = model && model.isPaged ? <PageSizeSelector model={model} /> : null;

        const exportBtn = model ? <Export model={model} advancedOption={advancedExportOption} /> : null;

        const chart =
            model && model.showChartSelector ? (
                <ChartSelector
                    model={model}
                    showSampleComparisonReports={showSampleComparisonReports}
                    onReportClicked={onReportClicked}
                    onCreateReportClicked={onCreateReportClicked}
                />
            ) : null;

        const view = model && model.showViewSelector ? <ViewSelector model={model} /> : null;

        let leftContent;

        if (buttons || chart) {
            leftContent = (
                <div className="col-md-6 col-sm-6 col-xs-12">
                    {buttonsNode}

                    {chart}
                </div>
            );
        }

        const rightContent = (
            <div className="col-md-6 col-sm-6 col-xs-12">
                <div className="paging pull-right text-nowrap">
                    {paging}
                    {pageSizeBtn}
                    {exportBtn}
                    {view}
                </div>
            </div>
        );

        return (
            <div className="query-grid-bar">
                <div className="row QueryGrid-bottom-spacing">
                    {leftContent}

                    {rightContent}
                </div>

                <div className="row QueryGrid-bottom-spacing">
                    <div className="col-md-12 col-xs-12">{box}</div>
                </div>

                <div className="row">
                    <div className="col-md-12 col-xs-12">
                        <GridSelectionBanner
                            containerCls="QueryGrid-bottom-spacing"
                            model={model}
                            onSelectionChange={onSelectionChange}
                        />
                    </div>
                </div>
            </div>
        );
    }
}
