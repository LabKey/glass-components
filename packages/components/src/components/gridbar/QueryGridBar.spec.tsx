import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import { SchemaQuery } from '../base/models/model';
import { initUnitTestMocks } from '../../testHelpers';
import { getStateQueryGridModel } from '../../models';
import { gridInit } from '../../actions';
import { getQueryGridModel } from '../../global';

import { QueryGridBar } from './QueryGridBar';

const SQ = new SchemaQuery({
    schemaName: 'assay.General.Amino Acids',
    queryName: 'Data',
});

beforeAll(() => {
    initUnitTestMocks();
});

describe('<QueryGridBar/>', () => {
    test('default props', done => {
        const model = getStateQueryGridModel('QueryGridBarDefaultProps', SQ, {});
        gridInit(model);

        window.setTimeout(() => {
            const component = <QueryGridBar model={getQueryGridModel(model.getId())} />;
            const tree = renderer.create(component);
            expect(tree.toJSON()).toMatchSnapshot();
            done();
        });
    });

    test('without charts and views', done => {
        const model = getStateQueryGridModel('QueryGridBarWithoutChartsViews', SQ, {
            showChartSelector: false,
            showViewSelector: false,
        });
        gridInit(model);

        window.setTimeout(() => {
            const component = <QueryGridBar model={getQueryGridModel(model.getId())} />;
            const tree = renderer.create(component);
            expect(tree.toJSON()).toMatchSnapshot();
            done();
        });
    });

    test('without omnibox', done => {
        const model = getStateQueryGridModel('QueryGridBarWithoutOmnibox', SQ, {
            showSearchBox: false,
        });
        gridInit(model);

        window.setTimeout(() => {
            const component = <QueryGridBar model={getQueryGridModel(model.getId())} />;
            const tree = renderer.create(component);
            expect(tree.toJSON()).toMatchSnapshot();
            done();
        });
    });

    test('isPaged', done => {
        const model = getStateQueryGridModel('QueryGridBarIsPaged', SQ, {
            isPaged: true,
        });
        gridInit(model);

        window.setTimeout(() => {
            const component = <QueryGridBar model={getQueryGridModel(model.getId())} />;
            const tree = renderer.create(component);
            expect(tree.toJSON()).toMatchSnapshot();
            done();
        });
    });
});
