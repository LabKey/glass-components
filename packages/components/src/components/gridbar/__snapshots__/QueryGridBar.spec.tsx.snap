// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<QueryGridBar/> default props 1`] = `
<div
  className="query-grid-bar"
>
  <div
    className="row QueryGrid-bottom-spacing"
  >
    <div
      className="col-md-6 col-sm-6 col-xs-12"
    >
      <span>
        <div
          className="dropdown disabled btn-group"
        >
          <button
            aria-expanded={false}
            aria-haspopup={true}
            className="dropdown-toggle btn btn-default"
            disabled={true}
            id="chartselector-0"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            type="button"
          >
            <span
              className="fa fa-spinner fa-pulse"
            />
             
            <span
              className="caret"
            />
          </button>
          <ul
            aria-labelledby="chartselector-0"
            className="dropdown-menu"
            role="menu"
          />
        </div>
      </span>
    </div>
    <div
      className="col-md-6 col-sm-6 col-xs-12"
    >
      <div
        className="paging pull-right text-nowrap"
      >
        <span
          className="gridbar-button-spacer"
        >
          <div
            className="dropdown btn-group"
          >
            <button
              aria-expanded={false}
              aria-haspopup={true}
              className="dropdown-toggle btn btn-default"
              disabled={false}
              id="export-drop-querygridbardefaultprops|assay$pgeneral$pamino acids/data"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
              role="button"
              type="button"
            >
              <span
                className="fa fa-download"
              />
            </button>
            <ul
              aria-labelledby="export-drop-querygridbardefaultprops|assay$pgeneral$pamino acids/data"
              className="dropdown-menu dropdown-menu-right"
              role="menu"
            >
              <li
                className="dropdown-header"
                onKeyDown={[Function]}
                role="heading"
              >
                Export 
                
              </li>
              <li
                className=""
                role="presentation"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="menuitem"
                  tabIndex="-1"
                >
                  <span
                    className="fa fa-file-o"
                  />
                    CSV
                </a>
              </li>
              <li
                className=""
                role="presentation"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="menuitem"
                  tabIndex="-1"
                >
                  <span
                    className="fa fa-file-excel-o"
                  />
                    Excel
                </a>
              </li>
              <li
                className=""
                role="presentation"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="menuitem"
                  tabIndex="-1"
                >
                  <span
                    className="fa fa-file-text-o"
                  />
                    TSV
                </a>
              </li>
            </ul>
          </div>
        </span>
        <span
          className="gridbar-button-spacer"
        >
          <div
            className="dropdown disabled btn-group"
          >
            <button
              aria-expanded={false}
              aria-haspopup={true}
              className="dropdown-toggle btn btn-default"
              disabled={true}
              id="viewselector-1"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
              type="button"
            >
              Grid Views
               
              <span
                className="caret"
              />
            </button>
            <ul
              aria-labelledby="viewselector-1"
              className="dropdown-menu dropdown-menu-right"
              role="menu"
            >
              <li
                className="active"
                role="presentation"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="menuitem"
                  tabIndex="-1"
                >
                  Default
                </a>
              </li>
            </ul>
          </div>
        </span>
      </div>
    </div>
  </div>
  <div
    className="row QueryGrid-bottom-spacing"
  >
    <div
      className="col-md-12 col-xs-12"
    >
      <div
        className="OmniBox OmniBox--multi"
      >
        <div
          className="OmniBox-control"
          onKeyDown={[Function]}
          onMouseDown={[Function]}
        >
          <span
            className="OmniBox-multi-value-wrapper"
          >
            <div
              className="OmniBox-placeholder"
            >
              Select...
            </div>
            <div
              className="OmniBox-input"
              style={
                Object {
                  "display": "inline-block",
                }
              }
            >
              <input
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                style={
                  Object {
                    "boxSizing": "content-box",
                    "width": "5pxpx",
                  }
                }
                value=""
              />
              <div
                style={
                  Object {
                    "height": 0,
                    "left": 0,
                    "overflow": "scroll",
                    "position": "absolute",
                    "top": 0,
                    "visibility": "hidden",
                    "whiteSpace": "pre",
                  }
                }
              >
                
              </div>
            </div>
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-md-12 col-xs-12"
    >
      <div
        className="QueryGrid-bottom-spacing"
      >
        <span
          className="QueryGrid-right-spacing"
        >
           
          <button
            className="btn btn-xs btn-default"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            Select all 
            34
          </button>
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`<QueryGridBar/> isPaged 1`] = `
<div
  className="query-grid-bar"
>
  <div
    className="row QueryGrid-bottom-spacing"
  >
    <div
      className="col-md-6 col-sm-6 col-xs-12"
    >
      <span>
        <div
          className="dropdown disabled btn-group"
        >
          <button
            aria-expanded={false}
            aria-haspopup={true}
            className="dropdown-toggle btn btn-default"
            disabled={true}
            id="chartselector-4"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            type="button"
          >
            <span
              className="fa fa-spinner fa-pulse"
            />
             
            <span
              className="caret"
            />
          </button>
          <ul
            aria-labelledby="chartselector-4"
            className="dropdown-menu"
            role="menu"
          />
        </div>
      </span>
    </div>
    <div
      className="col-md-6 col-sm-6 col-xs-12"
    >
      <div
        className="paging pull-right text-nowrap"
      >
        <span
          className="paging-counts-with-buttons"
          data-max={20}
          data-min={1}
          data-total={34}
        >
          <span>
            1
             - 
            20
          </span>
           
          of 
          34
        </span>
        <div
          className="btn-group"
        >
          <button
            className="disabled-button-with-tooltip btn btn-default"
            disabled={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
            type="button"
          >
            <i
              className="fa fa-chevron-left"
            />
          </button>
          <div
            className="dropdown btn-group"
          >
            <button
              aria-expanded={false}
              aria-haspopup={true}
              className="dropdown-toggle btn btn-default"
              disabled={false}
              id="current-page-drop-querygridbarispaged|assay$pgeneral$pamino acids/data"
              onClick={[Function]}
              onKeyDown={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
              role="button"
              type="button"
            >
              1
               
              <span
                className="caret"
              />
            </button>
            <ul
              aria-labelledby="current-page-drop-querygridbarispaged|assay$pgeneral$pamino acids/data"
              className="dropdown-menu dropdown-menu-right"
              role="menu"
            >
              <li
                className="dropdown-header"
                onKeyDown={[Function]}
                role="heading"
              >
                Jump To
              </li>
              <li
                className="disabled"
                role="presentation"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="menuitem"
                  tabIndex="-1"
                >
                  First Page
                </a>
              </li>
              <li
                className=""
                role="presentation"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="menuitem"
                  tabIndex="-1"
                >
                  Last Page
                </a>
              </li>
              <li
                className="dropdown-header"
                onKeyDown={[Function]}
                role="heading"
              >
                2
                 Total Pages
              </li>
            </ul>
          </div>
          <button
            className="btn btn-default"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
            type="button"
          >
            <i
              className="fa fa-chevron-right"
            />
          </button>
        </div>
        <span
          className="gridbar-button-spacer"
        >
          <div
            className="dropdown btn-group"
          >
            <button
              aria-expanded={false}
              aria-haspopup={true}
              className="dropdown-toggle btn btn-default"
              disabled={false}
              id="page-size-drop-querygridbarispaged|assay$pgeneral$pamino acids/data"
              onClick={[Function]}
              onKeyDown={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
              role="button"
              type="button"
            >
              20
               
              <span
                className="caret"
              />
            </button>
            <ul
              aria-labelledby="page-size-drop-querygridbarispaged|assay$pgeneral$pamino acids/data"
              className="dropdown-menu dropdown-menu-right"
              role="menu"
            >
              <li
                className="dropdown-header"
                onKeyDown={[Function]}
                role="heading"
              >
                Page Size
              </li>
              <li
                className="active"
                role="presentation"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="menuitem"
                  tabIndex="-1"
                >
                  20
                </a>
              </li>
              <li
                className=""
                role="presentation"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="menuitem"
                  tabIndex="-1"
                >
                  40
                </a>
              </li>
              <li
                className=""
                role="presentation"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="menuitem"
                  tabIndex="-1"
                >
                  100
                </a>
              </li>
              <li
                className=""
                role="presentation"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="menuitem"
                  tabIndex="-1"
                >
                  250
                </a>
              </li>
              <li
                className=""
                role="presentation"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="menuitem"
                  tabIndex="-1"
                >
                  400
                </a>
              </li>
            </ul>
          </div>
        </span>
        <span
          className="gridbar-button-spacer"
        >
          <div
            className="dropdown btn-group"
          >
            <button
              aria-expanded={false}
              aria-haspopup={true}
              className="dropdown-toggle btn btn-default"
              disabled={false}
              id="export-drop-querygridbarispaged|assay$pgeneral$pamino acids/data"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
              role="button"
              type="button"
            >
              <span
                className="fa fa-download"
              />
            </button>
            <ul
              aria-labelledby="export-drop-querygridbarispaged|assay$pgeneral$pamino acids/data"
              className="dropdown-menu dropdown-menu-right"
              role="menu"
            >
              <li
                className="dropdown-header"
                onKeyDown={[Function]}
                role="heading"
              >
                Export 
                
              </li>
              <li
                className=""
                role="presentation"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="menuitem"
                  tabIndex="-1"
                >
                  <span
                    className="fa fa-file-o"
                  />
                    CSV
                </a>
              </li>
              <li
                className=""
                role="presentation"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="menuitem"
                  tabIndex="-1"
                >
                  <span
                    className="fa fa-file-excel-o"
                  />
                    Excel
                </a>
              </li>
              <li
                className=""
                role="presentation"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="menuitem"
                  tabIndex="-1"
                >
                  <span
                    className="fa fa-file-text-o"
                  />
                    TSV
                </a>
              </li>
            </ul>
          </div>
        </span>
        <span
          className="gridbar-button-spacer"
        >
          <div
            className="dropdown disabled btn-group"
          >
            <button
              aria-expanded={false}
              aria-haspopup={true}
              className="dropdown-toggle btn btn-default"
              disabled={true}
              id="viewselector-5"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
              type="button"
            >
              Grid Views
               
              <span
                className="caret"
              />
            </button>
            <ul
              aria-labelledby="viewselector-5"
              className="dropdown-menu dropdown-menu-right"
              role="menu"
            >
              <li
                className="active"
                role="presentation"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="menuitem"
                  tabIndex="-1"
                >
                  Default
                </a>
              </li>
            </ul>
          </div>
        </span>
      </div>
    </div>
  </div>
  <div
    className="row QueryGrid-bottom-spacing"
  >
    <div
      className="col-md-12 col-xs-12"
    >
      <div
        className="OmniBox OmniBox--multi"
      >
        <div
          className="OmniBox-control"
          onKeyDown={[Function]}
          onMouseDown={[Function]}
        >
          <span
            className="OmniBox-multi-value-wrapper"
          >
            <div
              className="OmniBox-placeholder"
            >
              Select...
            </div>
            <div
              className="OmniBox-input"
              style={
                Object {
                  "display": "inline-block",
                }
              }
            >
              <input
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                style={
                  Object {
                    "boxSizing": "content-box",
                    "width": "5pxpx",
                  }
                }
                value=""
              />
              <div
                style={
                  Object {
                    "height": 0,
                    "left": 0,
                    "overflow": "scroll",
                    "position": "absolute",
                    "top": 0,
                    "visibility": "hidden",
                    "whiteSpace": "pre",
                  }
                }
              >
                
              </div>
            </div>
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-md-12 col-xs-12"
    >
      <div
        className="QueryGrid-bottom-spacing"
      >
        <span
          className="QueryGrid-right-spacing"
        >
           
          <button
            className="btn btn-xs btn-default"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            Select all 
            34
          </button>
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`<QueryGridBar/> without charts and views 1`] = `
<div
  className="query-grid-bar"
>
  <div
    className="row QueryGrid-bottom-spacing"
  >
    <div
      className="col-md-6 col-sm-6 col-xs-12"
    />
    <div
      className="col-md-6 col-sm-6 col-xs-12"
    >
      <div
        className="paging pull-right text-nowrap"
      >
        <span
          className="gridbar-button-spacer"
        >
          <div
            className="dropdown btn-group"
          >
            <button
              aria-expanded={false}
              aria-haspopup={true}
              className="dropdown-toggle btn btn-default"
              disabled={false}
              id="export-drop-querygridbarwithoutchartsviews|assay$pgeneral$pamino acids/data"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
              role="button"
              type="button"
            >
              <span
                className="fa fa-download"
              />
            </button>
            <ul
              aria-labelledby="export-drop-querygridbarwithoutchartsviews|assay$pgeneral$pamino acids/data"
              className="dropdown-menu dropdown-menu-right"
              role="menu"
            >
              <li
                className="dropdown-header"
                onKeyDown={[Function]}
                role="heading"
              >
                Export 
                
              </li>
              <li
                className=""
                role="presentation"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="menuitem"
                  tabIndex="-1"
                >
                  <span
                    className="fa fa-file-o"
                  />
                    CSV
                </a>
              </li>
              <li
                className=""
                role="presentation"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="menuitem"
                  tabIndex="-1"
                >
                  <span
                    className="fa fa-file-excel-o"
                  />
                    Excel
                </a>
              </li>
              <li
                className=""
                role="presentation"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="menuitem"
                  tabIndex="-1"
                >
                  <span
                    className="fa fa-file-text-o"
                  />
                    TSV
                </a>
              </li>
            </ul>
          </div>
        </span>
      </div>
    </div>
  </div>
  <div
    className="row QueryGrid-bottom-spacing"
  >
    <div
      className="col-md-12 col-xs-12"
    >
      <div
        className="OmniBox OmniBox--multi"
      >
        <div
          className="OmniBox-control"
          onKeyDown={[Function]}
          onMouseDown={[Function]}
        >
          <span
            className="OmniBox-multi-value-wrapper"
          >
            <div
              className="OmniBox-placeholder"
            >
              Select...
            </div>
            <div
              className="OmniBox-input"
              style={
                Object {
                  "display": "inline-block",
                }
              }
            >
              <input
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                style={
                  Object {
                    "boxSizing": "content-box",
                    "width": "5pxpx",
                  }
                }
                value=""
              />
              <div
                style={
                  Object {
                    "height": 0,
                    "left": 0,
                    "overflow": "scroll",
                    "position": "absolute",
                    "top": 0,
                    "visibility": "hidden",
                    "whiteSpace": "pre",
                  }
                }
              >
                
              </div>
            </div>
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-md-12 col-xs-12"
    >
      <div
        className="QueryGrid-bottom-spacing"
      >
        <span
          className="QueryGrid-right-spacing"
        >
           
          <button
            className="btn btn-xs btn-default"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            Select all 
            34
          </button>
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`<QueryGridBar/> without omnibox 1`] = `
<div
  className="query-grid-bar"
>
  <div
    className="row QueryGrid-bottom-spacing"
  >
    <div
      className="col-md-6 col-sm-6 col-xs-12"
    >
      <span>
        <div
          className="dropdown disabled btn-group"
        >
          <button
            aria-expanded={false}
            aria-haspopup={true}
            className="dropdown-toggle btn btn-default"
            disabled={true}
            id="chartselector-2"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            type="button"
          >
            <span
              className="fa fa-spinner fa-pulse"
            />
             
            <span
              className="caret"
            />
          </button>
          <ul
            aria-labelledby="chartselector-2"
            className="dropdown-menu"
            role="menu"
          />
        </div>
      </span>
    </div>
    <div
      className="col-md-6 col-sm-6 col-xs-12"
    >
      <div
        className="paging pull-right text-nowrap"
      >
        <span
          className="gridbar-button-spacer"
        >
          <div
            className="dropdown btn-group"
          >
            <button
              aria-expanded={false}
              aria-haspopup={true}
              className="dropdown-toggle btn btn-default"
              disabled={false}
              id="export-drop-querygridbarwithoutomnibox|assay$pgeneral$pamino acids/data"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
              role="button"
              type="button"
            >
              <span
                className="fa fa-download"
              />
            </button>
            <ul
              aria-labelledby="export-drop-querygridbarwithoutomnibox|assay$pgeneral$pamino acids/data"
              className="dropdown-menu dropdown-menu-right"
              role="menu"
            >
              <li
                className="dropdown-header"
                onKeyDown={[Function]}
                role="heading"
              >
                Export 
                
              </li>
              <li
                className=""
                role="presentation"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="menuitem"
                  tabIndex="-1"
                >
                  <span
                    className="fa fa-file-o"
                  />
                    CSV
                </a>
              </li>
              <li
                className=""
                role="presentation"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="menuitem"
                  tabIndex="-1"
                >
                  <span
                    className="fa fa-file-excel-o"
                  />
                    Excel
                </a>
              </li>
              <li
                className=""
                role="presentation"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="menuitem"
                  tabIndex="-1"
                >
                  <span
                    className="fa fa-file-text-o"
                  />
                    TSV
                </a>
              </li>
            </ul>
          </div>
        </span>
        <span
          className="gridbar-button-spacer"
        >
          <div
            className="dropdown disabled btn-group"
          >
            <button
              aria-expanded={false}
              aria-haspopup={true}
              className="dropdown-toggle btn btn-default"
              disabled={true}
              id="viewselector-3"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
              type="button"
            >
              Grid Views
               
              <span
                className="caret"
              />
            </button>
            <ul
              aria-labelledby="viewselector-3"
              className="dropdown-menu dropdown-menu-right"
              role="menu"
            >
              <li
                className="active"
                role="presentation"
              >
                <a
                  href="#"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="menuitem"
                  tabIndex="-1"
                >
                  Default
                </a>
              </li>
            </ul>
          </div>
        </span>
      </div>
    </div>
  </div>
  <div
    className="row QueryGrid-bottom-spacing"
  >
    <div
      className="col-md-12 col-xs-12"
    />
  </div>
  <div
    className="row"
  >
    <div
      className="col-md-12 col-xs-12"
    >
      <div
        className="QueryGrid-bottom-spacing"
      >
        <span
          className="QueryGrid-right-spacing"
        >
           
          <button
            className="btn btn-xs btn-default"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            Select all 
            34
          </button>
        </span>
      </div>
    </div>
  </div>
</div>
`;
